airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -374,11 +395,12 @@ def parse_response_error_message(cls, response: requests.Response) -> Optional[s
         """
 
         # default logic to grab error from common fields
-        def _try_get_error(value):
+        def _try_get_error(value: Optional[JsonType]) -> Optional[str]:
             if isinstance(value, str):
                 return value
             elif isinstance(value, list):
-                return ", ".join(_try_get_error(v) for v in value)
+                errors_in_value = [_try_get_error(v) for v in value]
+                return ", ".join(v for v in errors_in_value if v is not None)
```

Comment:
We probably never hit a None value here since it would've crashed, but the check is still warranted